src/Orleans/Storage/RelationalConstants.cs

```diff
@@ -0,0 +1,379 @@
+        /// </summary>
+        /// <remarks>Vendor specific.</remarks>
+        public const string DefaultConnectionStringKey = "DefaultConnectionString";
+
+        /// <summary>
+        /// The character that indicates a parameter.
+        /// </summary>
+        /// <remarks>Vendor specific.</remarks>
+        public const string ParameterIndicatorKey = "ParameterIndicator";
+
+        /// <summary>
+        /// The character that indicates a start escape key for columns and tables that are reserved words.
+        /// </summary>
+        /// <remarks>Vendor specific.</remarks>
+        public const string StartEscapeIndicatorKey = "StartEscapeIndicator";
+
+        /// <summary>
+        /// The character that indicates an end escape key for columns and tables that are reserved words.
+        /// </summary>
+        /// <remarks>Vendor specific.</remarks>
+        public const string EndEscapeIndicatorKey = "EndEscapeIndicator";
+
+        /// <summary>
+        /// A key for a query template to create a database with a given name.
+        /// </summary>
+        /// <remarks>Vendor specific.</remarks>
+        public const string CreateDatabaseKey = "CreateDatabaseTemplate";
+
+        /// <summary>
+        /// A key for a query template to drop a database with a given name.
+        /// </summary>
+        /// <remarks>Has vendor specific elements.</remarks>
+        public const string DropDatabaseKey = "DropDatabaseTemplate";
+
+        /// <summary>
+        /// A key for a query template to delete all data in tables.
+        /// </summary>
+        /// <remarks>Truncating would be more efficient, but is more vendor specific due to constraints.</remarks>
+        public const string DeleteAllDataKey = "DeleteAllDataTemplate";
+
+        /// <summary>
+        /// A key for a query template if a database with a given name exists.
+        /// </summary>
+        /// <remarks>Vendor specific.</remarks>
+        public const string ExistsDatabaseKey = "ExistsDatabaseTemplate";
+
+        /// <summary>
+        /// A collection of the well known keys Orleans uses to database management.
+        /// </summary>
+        public static IReadOnlyCollection<string> Keys = new ReadOnlyCollection<string>(new[]
+        {
+            NameKey,
+            DescriptionKey,
+            InvariantNameKey,
+            AssemblyQualifiedNameKey,
+            DefaultConnectionStringKey,
+            ParameterIndicatorKey,
+            StartEscapeIndicatorKey,
+            EndEscapeIndicatorKey,
+            CreateDatabaseKey,
+            DropDatabaseKey,
+            DeleteAllDataKey,
+            ExistsDatabaseKey,
+        });
+
+        /// <summary>
+        /// A parameter indicator that can be used in queries which is then substituted according to the
+        /// underlying database.
+        /// </summary>
+        public const string InvariantParameterIndicator = "@";
```

Comment:
When using reflection, the parameters need to be indicated somehow. Unfortunately the various databases have mostly their own indicators. Queries will work when using database specific indicators, but when using "invariant indicator", the queries should become vendor agnostic.